authorize_request_handler.go

```diff
@@ -363,27 +378,31 @@ func (f *Fosite) newAuthorizeRequest(ctx context.Context, r *http.Request, isPAR
 		return request, err
 	}
 
-	if err := f.validateAuthorizeRedirectURI(r, request); err != nil {
+	if err = f.parseAuthorizeScope(r, request); err != nil {
```

Comment:
I think that instead of splitting `validateAuthorizeScope` into `parseAuthorizeScope` and `validateAuthorizeScope`, it would be easier if you swapped the order of validation, first call (original) `validateAuthorizeScope` and then (updated) `validateAuthorizeRedirectURI`. In a way, if `validateAuthorizeRedirectURI` depends on parsed scopes, it is better if `validateAuthorizeRedirectURI` depends on parsed and validated scopes.